### DISCUSSION TITLE
How to use `join` and `form-control` `label` together

### DISCUSSION THREAD

**Discussion by longrunningprocess** [saadeghi/daisyui#2400] [Q&A] [answered]
I couldn't quite get the alignment right between the input and the button here (https://stackblitz.com/edit/vitejs-vite-vbebjm?file=src%2FApp.svelte) and I began to wonder if an example in the docs might be necessary or if this was possibly a missed consideration in the `join` work?

I was hoping for something along these lines as an API:

```html
<form role="search" class="form-control join">
   <label class="label join-item">
	<input type="search" name="q" bind:this={input} class="input input-lg input-bordered input-primary" />
	<span class="label-text-alt">'*' will return all concepts</span>
   </label>

   <button class="btn btn-primary btn-lg join-item">Search</button>
</form>
```

**Comment by longrunningprocess** [comment]
<img width="434" alt="image" src="https://github.com/saadeghi/daisyui/assets/4412848/6ede3ffc-5643-4c64-a6df-c473d5492112">

I think the focus ring might look better it were surrounding the entire "join group"...

**Comment by saadeghi** [comment, marked as answer]
`join-item` are for the items that should have automatic border radius, not essentially the direct children of `join`.  
`input` doesn't need to be inside `label`

Try this:  
https://play.tailwindcss.com/ERdlJ9tdXV
```html
<form role="search" class="join">
  
  <div class="form-control">
    <input type="search" name="q" bind:this="{input}" class="input join-item input-bordered input-primary input-lg" />
    <label class="label">
      <span class="label-text-alt">'*' will return all concepts</span>
    </label>
  </div>

  <button class="btn btn-primary join-item btn-lg">Search</button>
</form>
```

>I think the focus ring might look better it were surrounding the entire "join group"...

That may sound like a good idea at firrst but it would be confusing and would be an accessibility issue. There's an input and a button. Each one can bo focused independently. What would it mean if we apply focus ring around both of them? Which one is focused?

But anyway, if you want that style, you won't need `join`. Instead you can let input fill the width, and put the button on top of it like this:  
https://play.tailwindcss.com/P1NdEoELXw
```html
<form role="search" class="relative">
  
  <div class="form-control">
    <input type="search" name="q" bind:this="{input}" class="input input-bordered input-primary input-lg" />
    <label class="label">
      <span class="label-text-alt">'*' will return all concepts</span>
    </label>
  </div>

  <button class="btn btn-primary rounded-l-none btn-lg absolute right-0 top-0">Search</button>
</form>
```

**Comment by longrunningprocess** [comment]
> `join-item` are for the items that should have automatic border radius, not essentially the direct children of `join`. `input` doesn't need to be inside `label`

ah, ok... I was thinking too rigidly about that but that's an excellent clarification.  Thank you!

> 
> Try this: https://play.tailwindcss.com/ERdlJ9tdXV
> 
> ```
> <form role="search" class="join">
>   
>   <div class="form-control">
>     <input type="search" name="q" bind:this="{input}" class="input join-item input-bordered input-primary input-lg" />
>     <label class="label">
>       <span class="label-text-alt">'*' will return all concepts</span>
>     </label>
>   </div>
> 
>   <button class="btn btn-primary join-item btn-lg">Search</button>
> </form>
> ```

that's great, the HTML is still really clean for it, thanks for showing that 🌟

> 
> > I think the focus ring might look better it were surrounding the entire "join group"...
> 
> That may sound like a good idea at firrst but it would be confusing and would be an accessibility issue. There's an input and a button. Each one can bo focused independently. What would it mean if we apply focus ring around both of them? Which one is focused?
> 

oh yeah, I didn't think about it that much, I was getting too caught up on the looks, you're totally right.

> But anyway, if you want that style, you won't need `join`. Instead you can let input fill the width, and put the button on top of it like this: https://play.tailwindcss.com/P1NdEoELXw
> 
> ```
> <form role="search" class="relative">
>   
>   <div class="form-control">
>     <input type="search" name="q" bind:this="{input}" class="input input-bordered input-primary input-lg" />
>     <label class="label">
>       <span class="label-text-alt">'*' will return all concepts</span>
>     </label>
>   </div>
> 
>   <button class="btn btn-primary rounded-l-none btn-lg absolute right-0 top-0">Search</button>
> </form>
> ```

Thanks for the example but I agree with your observations, it wouldn't be appropriate...

As always, you are THE man!  🦸 💪🏻
